src/provider.toit

```diff
@@ -2,24 +2,62 @@
 // Use of this source code is governed by an MIT-style license that can be found
 // in the LICENSE file.
 
-import esp32
 import log
 import monitor
 import system.services show ServiceProvider ServiceResource ServiceHandler
 import .api.service
+import .esp32 show HardwareWatchdogEsp32
+
+interface HardwareWatchdog:
```

Comment:
I'm not sure hardware is a great name here given that this can be mocked, etc.

Reply:
Renamed back to "system" (`SystemWatchdog`)